### PR TITLE
Remove AWS from disable default providers until EKS plugin is fixed

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Due to a bug in upstream EKS plugin, we need to allow default AWS provider.
Update `test-infra` version to fix E2E tests due to Python 3.11

### Motivation

Bugfix.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ogaca-dd — APPROVED** [review]

File: test/new-e2e/go.mod [hunk 1]
```diff
@@ -16,7 +16,7 @@ require (
 	// Commit sha matches the first 8 digits of the commit sha in the module version
 	// Example: 	github.com/DataDog/test-infra-definitions v0.0.0-YYYYMMDDHHmmSS-0123456789AB
 	// => image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/test-infra-definitions/runner:01234567
-	github.com/DataDog/test-infra-definitions v0.0.0-20230619131020-fb813362e3e0
+	github.com/DataDog/test-infra-definitions v0.0.0-20230622123358-60eb38e90955
```
Comment (pducolin) [inline test/new-e2e/go.mod:19]: **💬 suggestion**
Will update the comment, did not know we can use the full commit sha as runner tag

File: test/new-e2e/utils/infra/stack_manager.go [hunk 2]
```diff
@@ -184,9 +184,14 @@ func buildWorkspace(ctx context.Context, profile runner.Profile, stackName strin
 		Description:    pulumi.StringRef("E2E Test inline project"),
 		StackConfigDir: stackName,
 		Config: map[string]workspace.ProjectConfigType{
-			// Always disable
+			// We should always disbale default providers
```
Comment (pducolin) [inline test/new-e2e/utils/infra/stack_manager.go:187]:
```suggestion
			// We should always disable default providers
```

**Review by pducolin — APPROVED** [review]

File: test/new-e2e/go.mod (hunk 1 above)
Comment (vboulineau) [reply test/new-e2e/go.mod:19]: I added dual tagging to be easier.

**Review by L3n41c — APPROVED** [review]

File: test/new-e2e/go.mod (hunk 1 above)
Comment (pducolin) [reply test/new-e2e/go.mod:19]: Nice !

**Pull request merged**